lib/rspec/core.rb

```diff
@@ -31,6 +31,7 @@
   option_parser
   configuration_options
   runner
+  profiler
```

Comment:
> Theres no two way communication required as long as you're ok with:

 > ```Ruby
config.profile_examples = true
config.profile_examples = false
```

> Not actually removing the profiler or the profile formatter.

I'm 100% OK with this.  In fact, I thought it was so obvious that I didn't think to mention it!

If the user toggles `profile_examples` from `true` to `false`, the correct user-facing behavior will still happen because we only notify the `dump_profile` event if the flag is set to `true`.  They'll still be paying the run-time cost of the profile stuff being tracked in memory but not output, but that's not a big deal, IMO.  The important things are that if `profile_examples` is set to true we produce the profile output and that if it is never set to `true` we don't waste time and resources loading and using the profiling logic.